### ISSUE TITLE
Shortcode in Paragraph blocks gets rendered only if there is another Shortcode block around.

### ISSUE THREAD

**Issue by dericleeyy** [WordPress/gutenberg#12449]
**Describe the bug**
When a shortcode (example `[youtube https://youtu.be/xifhQyopjZM]`) is placed in a Paragraph block, it is not rendered in the final page.

However, if a shortcode block is added to the page, the shortcode in the Paragraph block gets rendered as well.

**To Reproduce**
1. Start a new page
2. Add a paragraph block with content `[youtube https://youtu.be/xifhQyopjZM]`.
3. Publish
4. Edit the page. Add a shortcode block below the paragraph block with the same shortcode `[youtube https://youtu.be/xifhQyopjZM]`
5. Publish

**Expected behavior**
At step 3, the page displays the shortcode in text. At step 5, the page displays the shortcode in text with the embeded video below it.

**Actual Behaviour**
After step 5, the video is embedded twice on the page. The shortcode in the paragraph gets rendered.

**Desktop (please complete the following information):**
 - OS: [e.g. iOS] Mac OS 10.14.1
 - Browser [e.g. chrome, safari] Chrome
 - Version [e.g. 22] 70

**Additional context**
- Gutenberg 4.5.1
- WordPress 4.9.8
- Jetpack 6.8

**Comment by designsimply** [comment]
I tested with WordPress 4.9.8 and Gutenberg 4.6 and found that the shortcode in a paragraph does not get rendered as an embed unless Jetpack is also installed. In my testing, I found that this happens regardless of whether or not a shortcode block is also present.

Steps to reproduce:

1. Create a post with `[youtube https://youtu.be/xifhQyopjZM]` in a paragraph block.
1. Publish and view the post.
1. Install, activate, and connect Jetpack.
1. Refresh the view for the same post from earlier.

<strong>Result:</strong> shortcodes in paragraph blocks are not rendered on the front end with Gutenberg but are rendered on the front end when Jetpack is also activated and connected.

![screen shot 2018-11-30 at 9 42 38 am](https://user-images.githubusercontent.com/1119271/49302412-54c72d80-f484-11e8-844a-39f4719deb86.png)
<sup>Seen at https://tan-otter.jurassic.ninja/wp-admin/post.php?post=256&action=edit running WordPress 4.9.8 and Gutenberg 4.6 using Firefox 63.0.3 on macOS 10.13.6.</sup>
